src/main/java/com/minecolonies/coremod/util/InventoryUtils.java

```diff
@@ -294,603 +363,1248 @@ public static boolean hasToolLevel(final String tool, final ItemStack stack, fin
+     * Checked by {@link #getItemCountInProvider(ICapabilityProvider, Block,
+     * int)} &gt; 0;
+     *
+     * @param Provider   {@link ICapabilityProvider} to scan
+     * @param block      Block to count
+     * @param itemDamage the damage value.
+     * @return True when in {@link ICapabilityProvider}, otherwise false
+     */
+    public static boolean hasItemInProvider(@NotNull final ICapabilityProvider Provider, @NotNull final Block block, int itemDamage)
+    {
+        return hasItemInProvider(Provider, getItemFromBlock(block), itemDamage);
     }
 
     /**
-     * Assigns a string containing the grade of the toolGrade.
+     * Checks if a player has an item in the {@link ICapabilityProvider}.
+     * Checked by {@link #getItemCountInProvider(ICapabilityProvider, Item,
+     * int)} &gt; 0;
      *
-     * @param toolGrade the number of the grade of a tool
-     * @return a string corresponding to the tool
+     * @param Provider   {@link ICapabilityProvider} to scan
+     * @param item       Item to count
+     * @param itemDamage the damage value of the item.
+     * @return True when in {@link ICapabilityProvider}, otherwise false
      */
-    public static String swapToolGrade(final int toolGrade)
+    public static boolean hasItemInProvider(@NotNull final ICapabilityProvider Provider, @NotNull final Item item, int itemDamage)
     {
-        switch (toolGrade)
-        {
-            case 0:
-                return "Wood or Gold";
-            case 1:
-                return "Stone";
-            case 2:
-                return "Iron";
-            case 3:
-                return "Diamond";
-            default:
-                return "Better than Diamond";
-        }
+        return hasItemInProvider(Provider, (ItemStack stack) -> compareItems(stack, item, itemDamage));
     }
 
     /**
-     * Adapted from {@link net.minecraft.entity.player.InventoryPlayer#addItemStackToInventory(ItemStack)}.
+     * Checks if a player has an item in the {@link ICapabilityProvider}.
+     * Checked by {@link InventoryUtils#getItemCountInProvider(ICapabilityProvider,
+     * Predicate)} &gt; 0;
      *
-     * @param inventory Inventory to add itemstack to.
-     * @param itemStack ItemStack to add.
-     * @param building  the building.
-     * @return itemStack which has been replaced.
+     * @param Provider                    {@link ICapabilityProvider} to scan
+     * @param itemStackSelectionPredicate The predicate to match the ItemStack
+     *                                    to.
+     * @return True when in {@link ICapabilityProvider}, otherwise false
      */
-    @Nullable
-    public static ItemStack forceItemStackToInventory(@NotNull final IInventory inventory, @NotNull final ItemStack itemStack, @NotNull final AbstractBuilding building)
+    public static boolean hasItemInProvider(@NotNull final ICapabilityProvider Provider, @NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00117)